scripts/metrics/common/query_metrics.py

```diff
@@ -0,0 +1,157 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+
+Provides query functionality for .csv metrics from S3 buckets
+"""
+
+import argparse
+import csv
+import os
+
+from boto3.session import Session
+
+BUCKET_NAME = "o3de-metrics"
+
+SESSION = None
+DOWNLOADED_FILES = []
+
+
+def _get_session():
+    """
+    Gets an AWS session via boto3 or returns the existing one
+    """
+    global SESSION
+    if SESSION is None:
+        # get session
+        id_ = os.environ['AWS_ACCESS_KEY_ID']
```

Comment:
This will most likely need to be arguments to be passed in as a secret

Reply:
While it is reasonable to use something like getpass to pass a secret directly from a human user, doing so should not be necessary when running in AWS Lambda. I recommend looking at existing lambda scripts such as https://github.com/o3de/o3de/blob/development/scripts/build/lambda/trigger_first_build.py